pkgs/tools/security/otpgen/default.nix

```
@@ -22,7 +22,14 @@ stdenv.mkDerivation rec {
     sha256 = "19nyr3gmfxak0zdzibs1pjcdaxyy8x7jvrhgi3bsg9jbgf1m6xsr";
   };
 
-  patches = [ ./core-cmake-cxx.patch ./cmakelists.patch ];
+  # patches: 
+  #    - core-cmake-libcrypto++:
+  #       Fix on ./Source/Core/CMakeLists.txt, replacing libcrypto++ with 
+  #         libcryptopp.
+  #    - cmakelists-rpath;
+  #       Fix on ./CMakeLists.txt, removing unrecognized -rpath option.
+
+  patches = [ ./core-cmake-libcrypto++.patch ./cmakelists-rpath.patch ];
```

Formatting it like

```suggestion
  patches = [
    # Replacing libcrypto++ dependency with libcryptopp.
    ./core-cmake-libcrypto++.patch
    
    # Remove unrecognized -rpath option.
    ./cmakelists-rpath.patch  
  ];
```

is somewhat cleaner. Also it would be nice if you could add a link to upstream issues/pull requests fixing this in OTPgen.

---

Should I PR for the current latest HEAD of OTPGen or for the latest tag v0.9.3 ?

---

Formatting fixed here https://github.com/NixOS/nixpkgs/pull/87887/commits/6a6b782c3f8be9e47d9de021e141ce5b8fa1351d

---

I would assume OPTgen would want a pull request against their default branch.

This might help you to show them that `libcryptopp.pc` is the correct name:

https://github.com/weidai11/cryptopp/blob/b30600683178f613de9f72071d5d61bbb7ba0e0e/GNUmakefile#L1423-L1437

---

OTPGen PR for -rpath patch is here https://github.com/magiruuvelvet/OTPGen/pull/2.
PR for libcryptopp is here https://github.com/magiruuvelvet/OTPGen/pull/3.

Thanks for them (;

---

Added those PR links to the default.nix patch attribute as commentary https://github.com/NixOS/nixpkgs/pull/87887/commits/4cb9f5054616793ab4c9ae5e2e9460e3cba33a15